a1/README.md

```diff
@@ -3268,6 +3269,95 @@ Use [Gulp](http://gulpjs.com) or [Grunt](http://gruntjs.com) for creating automa
 
 **[Back to top](#table-of-contents)**
 
+## Low Level Functions
+
+Angular 1.x ships with some [helper functions](https://docs.angularjs.org/api/ng/function) that are found within JavaScript natively. In such cases we should favor usage of native JavaScript functions instead of Angular's helper functions. In other cases where functionality is not found in JavaScript natively, we should turn to utility libraries such as [lodash](https://lodash.com/docs) to fill-in the gaps.
+
+###### [Style [Y500](#style-y500)]
+
+  - Avoid using Angular low-level functions such as `angular.copy()`, `angular.extend()` or even `angular.isUndefined()`.
+
+  *Why?*: You can replace a lot Angular's low-level functions with native ES2015+ functionality such as [`Object.assign()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) for copying objects and the [Spread Syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator#Copy_an_array) for copying arrays _(ie. `[...arr1]`)_.
+
+  *Why?*: Because this is functionality that has been made available in the latest versions of JavaScript, but did not exist natively back when helper functions were first written.
+
+  Replace `angular.forEach()` and `angular.isArray()`:
+
+  ```javascript
+  /* avoid */
+  var arr = [1, 2, 3];
+  var obj = { hello: 'Hello', world: 'World' };
+
+  angular.forEach(arr, function(value) {
+    console.log(value);
+  });
+
+  angular.forEach(obj, function(value) {
+    console.log(value);
+  });
+
+  angular.isArray(arr); // true
+  angular.isArray(obj); // false
+  ```
+
+  ```javascript
+  /* recommended */
+  var arr = [1, 2, 3];
+  var obj = { hello: 'Hello', world: 'World' };
+
+  // iterate through arrays
+  arr.forEach(function(value) {
+    console.log(value);
+  });
+
+  // iterate through objects
+  Object.keys(obj)
+    .forEach(function(key) {
+      console.log(obj[key]);
+    });
+
+  // alternative lodash helper to iterate through objects
+  _.forEach(obj, function(value) {
+    console.log(value);
+  });
+
+  // available in ES
+  Array.isArray(arr); // true
+  Array.isArray(obj); // false
+  ```
+
+  Don't use `angular.copy()` and `angular.extend()` in favor of ES2015 features:
+
+  ```javascript
+  /* avoid */
+  var arr1 = [1, 2, 3];
+  var arr2 = angular.copy(arr1); // [1, 2, 3]
+
+  var obj1 = { hello: 'Hello' };
+  var obj2 = { world: 'World' };
+  var obj3 = angular.extend({}, obj1, obj2); // {hello: 'Hello', world: 'World'}
```

Comment:
Again, might not be a drop-in replacement, since `angular.extend` does for example preserve the destination's `$$hashKey`, which might result in different behavior in some cases. Not suggesting that such low-level details should be mentioned in the styleguide, just reinforcing the argument for adding a general note that - while you have to be aware of the fact that there might be slight differences - moving away from these helpers is the right thing to do (in many cases).